datalad_next/patches/push_to_export_remote.py

```diff
@@ -0,0 +1,240 @@
+import logging
+from typing import (
+    Dict,
+    Generator,
+    Iterable,
+    Optional,
+    Union,
+)
+from unittest.mock import patch
+
+import datalad.core.distributed.push as push
+from datalad.distribution.dataset import Dataset
+from datalad.runner.exception import CommandError
+from datalad.support.annexrepo import AnnexRepo
+from datalad.support.constraints import EnsureChoice
+from datalad.support.exceptions import CapturedException
+from datalad.support.param import Parameter
+from datalad_next.credman import CredentialManager
+from datalad_next.utils import (
+    get_specialremote_credential_envpatch,
+    get_specialremote_credential_properties,
+    needs_specialremote_credential_envpatch,
+)
+
+
+lgr = logging.getLogger('datalad.core.distributed.push')
+
+
+def _is_export_remote(remote_info: Optional[Dict]) -> bool:
+    """Check if remote_info is valid and has exporttree set to "yes"
+
+    Parameters
+    ----------
+    remote_info: Optional[Dict]
+        Optional dictionary the contains git annex special.
+
+    Returns
+    -------
+    bool
+        True if exporttree key is contained in remote_info and is set to yes,
+        else False.
+    """
+    if remote_info is not None:
+        return remote_info.get("exporttree") == "yes"
+    return False
+
+
+def _get_credentials(ds: Dataset,
+                     remote_info: Dict
+                     ) -> Optional[Dict]:
+
+    # Check for credentials
+    params = {
+        "type": remote_info.get("type"),
+        "url": remote_info.get("url")
+    }
+    credentials = None
+    credential_properties = get_specialremote_credential_properties(params)
+    if credential_properties:
+        # TODO: lower prio: factor this if clause out, also used in
+        #  create_sibling_webdav.py
+        credential_manager = CredentialManager(ds.config)
+        credentials = (credential_manager.query(
+            _sortby='last-used',
+            **credential_properties) or [(None, None)])[0][1]
+    return credentials
+
+
+# TODO: create a function similar to AnnexRepo.get_special_remote() and put it
+#  into core.
```

Comment:
Given that we already carry an `annexrepo` patch here, we can just actually patch it in, and register it in https://github.com/datalad/datalad/issues/6642

Reply:
done